paddle/fluid/operators/setitem_op.cc

```diff
@@ -0,0 +1,104 @@
+//   Copyright (c) 2020 PaddlePaddle Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "paddle/fluid/operators/setitem_op.h"
+
+#include <string>
+
+namespace paddle {
+namespace operators {
+
+class SetitemOp : public framework::OperatorWithKernel {
+ public:
+  SetitemOp(const std::string &type, const framework::VariableNameMap &inputs,
+            const framework::VariableNameMap &outputs,
+            const framework::AttributeMap &attrs)
+      : OperatorWithKernel(type, inputs, outputs, attrs) {}
+
+  void InferShape(framework::InferShapeContext *ctx) const override {
+    OP_INOUT_CHECK(ctx->HasInput("Input"), "Input", "Input", "Setitem");
+    OP_INOUT_CHECK(ctx->HasOutput("Out"), "Output", "Out", "Setitem");
+    auto in_dims = ctx->GetInputDim("Input");
+    PADDLE_ENFORCE_LT(
+        in_dims.size(), 7,
+        platform::errors::InvalidArgument(
+            "The rank of input should be less than 7, but received %d.",
```

Comment:
Why is this limit?

Reply:
Because `EigenTensor.slice` only supports Tensors with rank less than 7. And like [`slice` op](https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/fluid/operators/slice_op.cc#L61), here also limit the rank less than 7.

Reply:
我